openfifth/aspen-discovery
# Summon new features

This patch adds several new features to Summon, including the ability to control whether the Explore More Section is visible and to control whether available book covers are displayed when using the Summon Searcher.

Testing Book Covers:
This patch adds the ability to use only book covers generated by Aspen or to include those available from the API when carrying out a Summon Search. Choosing to use generated covers only increases the speed at which Summon loads results.

TEST PLAN:
(After applying patch).

1. Carry out a Summon Search as usual. You will notice that you are returned book covers that are generated by Aspen.
2. Navigate to Primary Configurations -> Library Systems and then edit whichever Library is your default.
3. Scroll to the Summon Section and check the tick box that is labelled, 'Show Available Covers in Summon.'
4. Click to save your changes and repeat your search.
5. Notice that the covers you are returned are now a mix of generated covers and actual book covers from the API - note that your results may load more slowly with this option enabled.

Testing Explore More Section:
This patch adds the ability to control whether the Explore More Section is visible in search results. By default, it is set to visible. 

TEST PLAN

1. Add the patch
2. Carry out a a Summon Search
3. Note that part way down your results list you will have a box labelled 'Explore More.'
4.  Navigate to the Layout Settings: Aspen Administration -> Theme and Layout -> Layout Settings.
5.  Within the layout settings, scroll to the checkbox labelled 'Show Explore More Options' and untick it.
6.  Repeat your search and notice that this time you do not have the 'Explore More' box in your results.
